RandBLAS/CMakeLists.txt

```diff
@@ -17,8 +17,15 @@ target_include_directories(RandBLAS INTERFACE
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/RandBLAS>)
 
-target_link_libraries(RandBLAS INTERFACE ${RandBLAS_libs})
-target_compile_options(RandBLAS INTERFACE -Wall -Wextra)
+target_link_libraries(RandBLAS PUBLIC ${RandBLAS_libs})
+
+set(RandBLAS_cxx_opts -Wall -Wextra)
+if (SANITIZE_ADDRESS)
+    list(APPEND RandBLAS_cxx_opts -fsanitize=address)
+    target_link_options(RandBLAS PUBLIC -fsanitize=address)
+endif()
+
```

Comment:
@burlen  CMake is angry that the ``INTERFACE`` specifications in ``target_link_libraries`` and ``target_compile_options`` has been replaced with ``PUBLIC``. It looks like changing these back to ``INTERFACE`` fixes things.

Reply:
yes, that was because of the change to header only (use INTERFACE) from a compiled library(use PUBLIC)